src/test/ui/borrowck/borrowck-issue-48962.stderr

```diff
@@ -1,3 +1,9 @@
+error[E0718]: cannot assign to `src.next` when `src` is not initialized
+  --> $DIR/borrowck-issue-48962.rs:26:5
+   |
+LL |     src.next = None; //~ ERROR use of moved value: `src` [E0382]
```

Comment:
(I've already chatted with @spastorino about this; we can avoid this (IMO redundant/unnecessary) error if we revise the strategy slightly.  PR #54941 has some hidden suggestions )